modules/develop/pages/data-transforms/deploy.adoc

```diff
@@ -0,0 +1,95 @@
+= Deploy Data Transforms
+:description: Learn how to build, deploy, share, and troubleshoot data transforms in Redpanda.
+:page-categories: Development, Stream Processing, Data Transforms
+
+{description}
+
+== Prerequisites
+
+Before you begin, ensure that you have the following:
+
+- xref:develop:data-transforms/configure.adoc#enable-transforms[Data transforms enabled] in your Redpanda cluster.
+- The xref:get-started:rpk-install.adoc[`rpk` command-line client] installed on your host machine and configured to connect to your Redpanda cluster.
+- A xref:develop:data-transforms/build.adoc[data transform] project.
+
+[[build]]
+== Build the Wasm binary
+
+To build a Wasm binary:
+
+1. Ensure your project directory contains a `transform.yaml` file.
+2. Build the Wasm binary using the xref:reference:rpk/rpk-transform/rpk-transform-build.adoc[`rpk transform build`] command.
+
+[source,bash]
+----
+rpk transform build
+----
+
+You should now have a Wasm binary named `<transform-name>.wasm`, where `<transform-name>` is the name specified in your `transform.yaml` file. This binary is your data transform function, ready to be deployed to a Redpanda cluster or hosted on a network for others to use.
+
+[[deploy]]
+== Deploy the Wasm binary
+
+You can deploy your transform function using the xref:reference:rpk/rpk-transform/rpk-transform-deploy.adoc[`rpk transform deploy`] command.
+
+. Validate your setup against the pre-deployment checklist:
++
+[%interactive]
+- [ ] Do you meet the prerequisites?
+- [ ] Do you have a Wasm binary located in the project directory?
+- [ ] Is the `transform.yaml` file correctly configured?
+- [ ] Does your transform function require any environment variables? If so, make sure to set them in the `transform.yaml` file or in the command-line when you deploy the binary.
+- [ ] Do your input and output topics already exist? Input and output topics must exist in your Redpanda cluster before you deploy the Wasm binary.
+
+. Deploy the Wasm binary:
++
+[source,bash]
+----
+rpk transform deploy --file=<path-to-wasm-binary> --var <environment-variable-name>=<value>
```

Comment:
You don't need the `file` flag or the `var` flag if you're using transform.yaml

I recommend everyone use a `transform.yaml` file, but if you want to override a value on the CLI, you can do that (they get merged), or you can do all flags and drop the config file.